v2/tests/git_initializer_test.go

```diff
@@ -0,0 +1,201 @@
+// +build integration
+
+package tests
+
+import (
+	"context"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/brigadecore/brigade/sdk/v2"
+	"github.com/brigadecore/brigade/sdk/v2/authx"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+	"github.com/stretchr/testify/require"
+)
+
+var defaultConfigFiles = map[string]string{
+	"brigade.js": `
+		const { events } = require("brigadier");
+
+		events.on("github.com/brigadecore/brigade/cli", "exec", () => {
+			console.log("Hello, World!");
+		});
+	`}
+
+var testcases = []struct {
+	name          string
+	project       core.Project
+	terminalPhase core.WorkerPhase
+}{
+	{
+		name: "GitHub - no ref",
+		project: core.Project{
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						CloneURL: "https://github.com/brigadecore/empty-testbed.git",
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	{
+		name: "GitHub - full ref set",
+		project: core.Project{
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						CloneURL: "https://github.com/brigadecore/empty-testbed.git",
+						Ref:      "refs/heads/master",
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	{
+		name: "GitHub - short ref set",
+		project: core.Project{
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						CloneURL: "https://github.com/brigadecore/empty-testbed.git",
+						Ref:      "master",
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	{
+		name: "GitHub - submodules",
+		project: core.Project{
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						// TODO: host similar repo in brigadecore
+						CloneURL:       "https://github.com/sgoings/makeup.git",
+						InitSubmodules: true,
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseSucceeded,
+	},
+	{
+		name: "GitHub - non-existent ref",
+		project: core.Project{
+			Spec: core.ProjectSpec{
+				WorkerTemplate: core.WorkerSpec{
+					Git: &core.GitConfig{
+						CloneURL: "https://github.com/brigadecore/empty-testbed.git",
+						Ref:      "non-existent",
+					},
+				},
+			},
+		},
+		terminalPhase: core.WorkerPhaseFailed,
+	},
```

Comment:
I just noticed that we're missing a test case that covers using a commit sha instead of a ref.

Reply:
Good call; added.